main-service/src/main/java/ru/practicum/ewm/service/event/EventPublicService.java

```diff
@@ -58,37 +56,36 @@ public EventFullDto getEvent(Long eventId, HttpServletRequest request) {
         return eventFullDto;
     }
 
-
-    @Transactional(readOnly = true)
-    public List<EventFullDto> search(EventPublicSearch ev, String ip) {
+    public List<EventFullDto> search(EventSearch ev, String ip) {
         if (ev == null) {
             return Collections.emptyList();
         }
-        List<Event> events = eventRepository.publicSearch(ev.getText(), ev.getCategories(), ev.getPaid(),
-                ev.getRangeStart(), ev.getRangeEnd(), ev.getFrom(), ev.getSize());
+        List<Event> events = eventRepository.search(ev);
         if (events.isEmpty()) {
             return Collections.emptyList();
         }
-        List<EventFullDto> eventFullDtos = events.stream()
-                .map(EventMapper::eventToEventFullDto).collect(Collectors.toList());
+        List<EventFullDto> eventFullDtos = events.stream().map(EventMapper::eventToEventFullDto).collect(Collectors.toList());
         Map<Long, Long> views = eventCommonService.getStats(events, false);
         eventFullDtos.forEach(e -> e.setViews(views.get(e.getId())));
+        eventCommonService.setLikesToEventDtos(events, eventFullDtos);
         List<ParticipationRequest> confirmedRequests = requestService.findConfirmedRequests(events);
-        for (EventFullDto eventFullDto : eventFullDtos) {
-            eventFullDto.setConfirmedRequests((int) confirmedRequests.stream()
-                    .filter(r -> r.getEvent().getId().equals(eventFullDto.getId())).count());
-        }
+        Map<Long, Long> eventRequestsCount = confirmedRequests.stream().collect(Collectors.groupingBy(r -> r.getEvent().getId(), Collectors.counting()));
+        eventFullDtos.forEach(e -> e.setConfirmedRequests(eventRequestsCount.get(e.getId())));
         HitDto hitDto = HitDto.builder().app("APP_NAME").uri("/events").ip(ip).timestamp(LocalDateTime.now().withNano(0)).build();
         statsClient.createHit(hitDto);
-        events.forEach(e -> {
-            hitDto.setUri("/events/" + e.getId());
-            statsClient.createHit(hitDto);
-        });
-        if (ev.getSort() != null && ev.getSort().equals(Sort.VIEWS)) {
-            return eventFullDtos.stream()
-                    .sorted(Comparator.comparing(EventFullDto::getViews)).collect(Collectors.toList());
+        if (ev.getSort() != null) {
```

Comment:
....................     
return eventFullDtos.stream()
                .sorted(getComparator(ev.getSort())).collect(Collectors.toList());
    }
    
    private Comparator<EventFullDto> getComparator(Sort sort) {
        switch (sort) {
            case VIEWS:
                return Comparator.comparing(EventFullDto::getViews);
            case RATING:
                return Comparator.comparing(EventFullDto::getLikesCount);
            case EVENT_DATE:
                return Comparator.comparing(EventFullDto::getEventDate);
            default:
                throw new IllegalArgumentException(String.format("Sort %s is unknown", sort));
        }
    }